### PR TITLE
Fix/select all

### PR DESCRIPTION
**Pull request by saadshahd**
Changes:
- Adds `[contenteditable="true"]` to the editor
- Removes the `[contenteditable]` property from all the components
- Adds `[contenteditable="false"]` to `<figure>` media element

### CODE REVIEW

**Comment by saadshahd** [comment]
Hey @mkhatib, Please check this PR as a fix for #80

**Comment by mkhatib** [comment]
Thanks for sending this out, there was a reason I switched form a single content editable on the whole editor and stuck with content editable from other components.

I can't remember what was it exactly, I'll try to run this locally and see if I can find any problems.

File: src/extensions/giphyComponent.js [hunk 1]
```diff
@@ -220,7 +220,6 @@ GiphyComponent.prototype.render = function(element, options) {
       this.selectionDom = document.createElement('div');
       this.selectionDom.innerHTML = '&nbsp;';
       this.selectionDom.className = 'selection-pointer';
-      this.selectionDom.setAttribute('contenteditable', true);
```
Comment (mkhatib) [inline src/extensions/giphyComponent.js:223]: This change is breaking the selection process of the elements (including `FigureComponent`). 

Try for example clicking on an image and notice that there's a cursor in the paragraph before hitting delete won't delete the image and instead would delete the character in previous paragraph.

File: src/layout.js [hunk 2]
```diff
@@ -28,7 +28,6 @@ var Layout = function(optParams) {
   Section.call(this, params);
 
   this.type = params.type;
-  this.dom.setAttribute('contenteditable', false);
```
Comment (mkhatib) [inline src/layout.js:31]: This is also a breaking change. This allows the cursor to be inside layout container. You can see this is breaking by staging an image and then sitting the cursor at a paragraph after that stage and click the back arrow to see the cursor moving into that layout and user would be able to type outside the editing area.

**Review by mkhatib — CHANGES_REQUESTED** [review]
@saadshahd tried to play with this locally and it breaks quiet few things. The reason I believe I did the `contenteditable` change was because cursors would be stuck in the wrong places between the `divs` `containers` and `layout`. 

If we can make sure that doesn't happen I am ok with changing the `contenteditable` situation. 

Let me know if you can do some more testing on this and see how to fix these issues.

**Comment by mkhatib** [comment]
Thanks @saadshahd I've used the same methods you were using here in the other PR to fix this. Thanks for taking a stab at this 😄

**Pull request closed without merging**